data-prepper-plugins/sns-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/SNSSinkServiceTest.java

```diff
@@ -0,0 +1,139 @@
+            "        aws:\n" +
+            "          region: ap-south-1\n" +
+            "          sts_role_arn: arn:aws:iam::524239988912:role/app-test\n" +
+            "          sts_header_overrides: {\"test\":\"test\"}\n" +
+            "        codec:\n" +
+            "          ndjson:\n" +
+            "        max_retries: 10\n" +
+            "        dlq_file: /test/dlq-file.log\n" +
+            "        dlq:\n" +
+            "          s3:\n" +
+            "            bucket: test\n" +
+            "            key_path_prefix: test\n";
+    private PublishBatchResponse publishBatchResponse;
+
+    private SdkHttpResponse sdkHttpResponse;
+
+
+    @BeforeEach
+    public void setup() throws JsonProcessingException {
+        this.snsSinkConfig = objectMapper.readValue(config,SNSSinkConfig.class);
+        this.snsClient = mock(SnsClient.class);
+        this.pluginMetrics = mock(PluginMetrics.class);
+        this.numberOfRecordsSuccessCounter = mock(Counter.class);
+        this.numberOfRecordsFailedCounter = mock(Counter.class);
+        this.publishBatchResponse = mock(PublishBatchResponse.class);
+        this.sdkHttpResponse = mock(SdkHttpResponse.class);
+
+        when(pluginMetrics.counter(SNSSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_SUCCESS)).thenReturn(numberOfRecordsSuccessCounter);
+        when(pluginMetrics.counter(SNSSinkService.NUMBER_OF_RECORDS_FLUSHED_TO_SNS_FAILED)).thenReturn(numberOfRecordsFailedCounter);
+        when(snsClient.publishBatch(any(PublishBatchRequest.class))).thenReturn(publishBatchResponse);
+        when(publishBatchResponse.sdkHttpResponse()).thenReturn(sdkHttpResponse);
+        when(publishBatchResponse.sdkHttpResponse().statusCode()).thenReturn(new Random().nextInt());
+        when(sdkHttpResponse.statusCode()).thenReturn(200);
+
+    }
+
+
+    private SNSSinkService createObjectUnderTest(){
+        return new SNSSinkService(snsSinkConfig,
+                snsClient,
+                pluginMetrics, mock(PluginFactory.class), mock(PluginSetting.class));
+    }
+
+    @Test
+    public void sns_sink_test_with_empty_collection_records(){
+        numberOfRecordsSuccessCounter = mock(Counter.class);
+        SNSSinkService snsSinkService = createObjectUnderTest();
+        snsSinkService.output(List.of());
+        verifyNoInteractions(numberOfRecordsSuccessCounter);
+        verifyNoInteractions(numberOfRecordsFailedCounter);
+    }
+
+    @Test
+    public void sns_sink_test_with_single_collection_record_success_push_to_sns(){
+        when(sdkHttpResponse.statusCode()).thenReturn(500);
+        SNSSinkService snsSinkService = createObjectUnderTest();
+        final Record<Event> eventRecord = new Record<>(JacksonEvent.fromMessage("{\"message\":\"c3f847eb-333a-49c3-a4cd-54715ad1b58a\"}"));
+        Collection<Record<Event>> records = List.of(eventRecord);
+        snsSinkService.output(records);
+        verify(numberOfRecordsSuccessCounter).increment(records.size());
+    }
+
```

Comment:
Please add a test which publishes batchSize (e.g. 10) + 1 records. It should verify that publishBatch was called twice. The first call with batchSize records in the first call, and 1 record in the second.

Reply:
Thanks @dlvenable for your feedback, Sure, it will available in next commit for review.